src/lib.rs

```diff
@@ -108,6 +192,17 @@ where
 {
     let rsdp_mapping = handler.map_physical_region::<Rsdp>(rsdp_address, mem::size_of::<Rsdp>());
     (*rsdp_mapping).validate()?;
+
+    parse_validated_rsdp(handler, rsdp_mapping)
+}
+
+fn parse_validated_rsdp<H>(
```

Comment:
Not convinced we should be doing the validation elsewhere. If not, this can move back into `parse_rsdp`

Reply:
Final comments on this? Are we going to leave it like this or not considering what we've discussed on gitter?

Reply:
Yeah, I think this is fine. It’s difficult to understand the need for the split without understanding the nuances of the RSDP search, but that’s probably fine given it’s internal.

Reply:
Well, the split here is just so that we don't waste 4kb of memory which is the search area chunk. If we just called the normal method it would descend into the whole acpi crate and only free the 4kb at the end.